### PR TITLE
ENH use openblas too

### LINKED ISSUE #101: BLAS pinned to MKL in x64 is problematic in macOS rosetta environments
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

Pytorch cannot be installed in a macOS rosetta environment emulating x64 because our recipe pins BLAS to MKL (for example, **[here](https://github.com/conda-forge/pytorch-cpu-feedstock/blob/93333de760d702aee276dcafeedc890458a00417/recipe/meta.yaml#L93-L94))**. In turn, MKL refuses to run in a rosetta environment after some runtime checks.

To reproduce, in an M1 mac machine and an x86_64 conda installation, run:
```sh
conda create -n pytorch-test -c conda-forge pytorch
```

To fix this, we would need to build against other BLAS option and allow to use it in rosetta environments. But I do not know if there is a conda selector or other mechanism we could use to detect that is the case.

As a workaround we can pin MKL to version <2022, which does not seem to include the runtime check. I do not know if this can lead to problems further down the line.

### Installed packages

```shell
-
```

### Environment info

```shell
-
```

### LINKED ISSUE #174: build without mkl support on linux?
We should be able to build without mkl support on linux and make those packages lower priority. If I make the PR, are you all interested in that change?

cc @conda-forge/pytorch-cpu @conda-forge/core

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

closes #174 
fixes #101

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
@conda-forge-admin restart ci

File: recipe/build_pytorch.sh [hunk 1]
```diff
@@ -74,6 +74,14 @@ fi
 
 export MAX_JOBS=${CPU_COUNT}
 
+if [[ "$blas_impl" == "openblas" ]]; then
+    export BLAS=OpenBLAS
+    export USE_MKLDNN=0
+else
+    export BLAS=mkl
```
Comment (beckermr) [inline recipe/build_pytorch.sh:81]:
```suggestion
    export BLAS=MKL
```

**Comment by hmaarrfk** [comment]
would it be ok to skip a few builds (py!=10) while things get debugged?

**Comment by beckermr** [comment]
Wow. Smithy is broken or IDK what I am doing or both. When I add skip for `py != 310` it doesn't build anything.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -18,20 +18,22 @@ source:
...
 
 build:
-  number: {{ number }}
+  number: {{ number + 100 }}  # [blas_impl == "mkl"]
+  number: {{ number }}  # [blas_impl != "mkl"]
   skip: true  # [win]
   skip: true  # [cuda_compiler_version == "10.2"]
   skip: true  # [cuda_compiler_version == "11.0"]
   skip: true  # [cuda_compiler_version == "11.1"]
+  skip: true  # [py != 310]
```
Comment (isuruf) [inline recipe/meta.yaml:30]: move this to below line 37

Comment (isuruf) [reply recipe/meta.yaml:30]: and copy to line 165

Comment (beckermr) [reply recipe/meta.yaml]: This didn't appear to help.

Comment (beckermr) [reply recipe/meta.yaml:30]: Adding a top-level requirements section with python did the trick. I wonder if this is a smithy bug or conda build bug?

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

File: recipe/meta.yaml [hunk 3]
```diff
@@ -79,8 +88,12 @@ outputs:
         - requests
         - future
         - six
-        - mkl-devel {{ mkl }}   # [x86]
-        - libcblas * *_mkl      # [x86]
+        - mkl-devel {{ mkl }}   # [x86 and blas_impl == "mkl"]
+        - libcblas * *_mkl      # [x86 and blas_impl == "mkl"]
+        # - onednn * omp_*        # [x86 and blas_impl == "mkl"]
```
Comment (beckermr) [inline recipe/meta.yaml:93]:
```suggestion
          - onednn * omp_*        # [x86 and blas_impl == "mkl"]
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -79,8 +88,12 @@ outputs:
         - requests
         - future
         - six
-        - mkl-devel {{ mkl }}   # [x86]
-        - libcblas * *_mkl      # [x86]
+        - mkl-devel {{ mkl }}   # [x86 and blas_impl == "mkl"]
+        - libcblas * *_mkl      # [x86 and blas_impl == "mkl"]
+        - onednn * omp_*        # [x86 and blas_impl == "mkl"]
```
Comment (beckermr) [inline recipe/meta.yaml:93]:
```suggestion
        - onednn * omp_*
```

**Comment by hmaarrfk** [comment]
How do we want to merge this with the progress on 2.0.1?

I can work to collapse my commits.
https://github.com/conda-forge/pytorch-cpu-feedstock/pull/176

**Comment by beckermr** [comment]
I don't want to break your existing workflow at all with this since it doesn't appear to be well supported upstream anyways.

File: recipe/meta.yaml [hunk 5]
```diff
@@ -79,8 +88,12 @@ outputs:
         - requests
         - future
         - six
-        - mkl-devel {{ mkl }}   # [x86]
-        - libcblas * *_mkl      # [x86]
+        - mkl-devel {{ mkl }}   # [x86 and blas_impl == "mkl"]
+        - libcblas * *_mkl      # [x86 and blas_impl == "mkl"]
+        # - onednn * omp_*
```
Comment (beckermr) [inline recipe/meta.yaml:93]:
```suggestion
```

**Comment by hmaarrfk** [comment]
well i can't get 2.0.1 to actually import with cuda. so i'm glad you are pushing forward here.

File: recipe/meta.yaml [hunk 6]
```diff
@@ -18,23 +18,32 @@ source:
...
     # We've patch out the bug in GCC that makes it incompatible
     # https://github.com/conda-forge/ctng-compilers-feedstock/blob/7ea10c2526445beb9faff23efcc6bc40fc0079b5/recipe/meta.yaml#L23=
-    - patches/remove_gcc_maximum_requirements.patch 
+    - patches/remove_gcc_maximum_requirements.patch
 
 build:
-  number: {{ number }}
+  number: {{ number + 100 }}  # [blas_impl == "mkl"]
+  number: {{ number }}  # [blas_impl != "mkl"]
   skip: true  # [win]
```
Comment (isuruf) [inline recipe/meta.yaml:26]: If the user doesn't mind cuda, I guess they wouldn't mind MKL too?
```suggestion
  skip: true  # [cuda_compiler_version != "None" and blas_impl != "mkl"]
```

Comment (beckermr) [reply recipe/meta.yaml:26]: That seems fine but wouldn't this add windows builds back?

Comment (isuruf) [reply recipe/meta.yaml:26]: Fixed

Comment (beckermr) [reply recipe/meta.yaml:26]: Thank you!

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
Hmmm @isuru maybe we need skips in other spots?

File: recipe/meta.yaml [hunk 7]
```diff
@@ -104,6 +113,7 @@ outputs:
         - filelock
         - jinja2
         - networkx
+        - nomkl                 # [blas_impl == "mkl"]
```
Comment (beckermr) [inline recipe/meta.yaml:116]: nomkl with blas_impl == mkl?

File: recipe/meta.yaml [hunk 8]
```diff
@@ -18,20 +18,29 @@ source:
...
 build:
-  number: {{ number }}
+  number: {{ number + 100 }}  # [blas_impl == "mkl"]
+  number: {{ number }}  # [blas_impl != "mkl"]
   skip: true  # [win]
+  skip: true  # [cuda_compiler_version != "None" and blas_impl != "mkl"]
   skip: true  # [cuda_compiler_version == "10.2"]
   skip: true  # [cuda_compiler_version == "11.0"]
   skip: true  # [cuda_compiler_version == "11.1"]
+  skip: true  # [py != 310]
```
Comment (beckermr) [inline recipe/meta.yaml:31]:
```suggestion
```

File: recipe/meta.yaml [hunk 9]
```diff
@@ -18,20 +18,29 @@ source:
...
+
+requirements:
+  host:
+    - python
+  run:
+    - python
```
Comment (beckermr) [inline recipe/meta.yaml:37]:
```suggestion
```

**Review by beckermr — COMMENTED** [review]
A bit more python 10 clean up.

**Comment by beckermr** [comment]
CUDA timed out but otherwise this looks ready to go. @rgommers you wanted to do some testing here outside of CI. Any suggestions on what to run? I don't have direct access to a linux box but can try to dig something up.

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
@conda-forge-admin rerender

File: recipe/meta.yaml [hunk 10]
```diff
@@ -18,20 +18,28 @@ source:
...
   skip: true  # [cuda_compiler_version == "10.2"]
   skip: true  # [cuda_compiler_version == "11.0"]
   skip: true  # [cuda_compiler_version == "11.1"]
 
+requirements:
+  host:
+    - python
+  run:
+    - python
+
```
Comment (beckermr) [inline recipe/meta.yaml:37]:
```suggestion
```

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
Ok this is ready to merge I think.  Comments?

**Comment by rgommers** [comment]
> CUDA timed out but otherwise this looks ready to go. @rgommers you wanted to do some testing here outside of CI. Any suggestions on what to run? I don't have direct access to a linux box but can try to dig something up.

My suggestion was the test suite, for which the entry point was [test/run_test.py](https://github.com/pytorch/pytorch/blob/main/test/run_test.py). I have a Linux + CUDA dev machine, so I may be able to help. This week is super busy unfortunately, and I'm not sure how to get a conda env with a built package from this PR. If it's easier to do post-merge and fix things up retroactively in case something doesn't work, that's also fine I'd say.

**Comment by beckermr** [comment]
Neither our current MKL builds or the new openblas builds actually pass the test suite.

**Comment by beckermr** [comment]
Given that I think we should merge and see what bubbles up.

**Comment by beckermr** [comment]
Bump here @isuruf @rgommers 

Shall we merge?

**Comment by rgommers** [comment]
If the set of failures looks roughly similar, then sure go ahead. I'll see if I can have a look at what does and doesn't pass.

**Comment by beckermr** [comment]
> If the set of failures looks roughly similar, then sure go ahead. I'll see if I can have a look at what does and doesn't pass.

That makes sense. The failures are not identical for sure. 

One question. I cannot figure out how to get a succinct list of the failing tests out of the test suite. Any help on the right commands to run would be appreciated.

**Comment by beckermr** [comment]
LOL

Both builds segfault during the test suite and the openblas one fails a few extra tests not failed by the MKL one.

openblas:

<details/>

```bash
============================= test session starts ==============================
platform linux -- Python 3.10.12, pytest-7.4.0, pluggy-1.2.0
rootdir: /direct/astro+u/beckermr/pytorch
configfile: pytest.ini
plugins: hypothesis-6.82.0
collected 192953 items / 60 errors / 1 skipped

run_test.py EEEE                                                         [  0%]
test_ao_sparsity.py .[W qlinear_dynamic.cpp:247] Warning: Currently, qnnpack incorrectly ignores reduce_range when it is set to true; this may change in a future release. (function operator())
................................................... [  0%]
.................................                                        [  0%]
test_autocast.py .......s.                                               [  0%]
test_autograd.py .................................s.s......sss.......... [  0%]
........................................................................ [  0%]
............s..............................x............................ [  0%]
....F.F................................................................. [  0%]
........................................................................ [  0%]
......................ss................................................ [  0%]
.....................ss................................................. [  0%]
.s.s....ss.s...............s....sss.s..s........s.......s.....ss...      [  0%]
test_binary_ufuncs.py .................................................. [  0%]
........................................................................ [  0%]
........................................................................ [  0%]
........................................................................ [  0%]
........................................................................ [  0%]
..................s..................................................... [  0%]
........................................................................ [  0%]
........................................................................ [  0%]
.......................s................................................ [  0%]
...............................................ssssssssssssssssssssssss. [  0%]
........................................................................ [  0%]
........................................................................ [  0%]
................................................ssssssssssssssss........ [  0%]
................ssssssssssssssssssssssssssssssssssssssss................ [  0%]
.....................................s.................................. [  0%]
ssss.................................................................... [  0%]
.......................................F................................ [  0%]
........................................................................ [  0%]
........................................................................ [  1%]
.....................................................s.................. [  1%]
........................................................................ [  1%]
.....ssssssssssssssssssssssss........................................... [  1%]
........................................................................ [  1%]
........................................................................ [  1%]
...............................s........................................ [  1%]
.......................................................sssssssssssssssss [  1%]
sssssss................................................................. [  1%]
........................................................................ [  1%]
........................................................ssssssssssssssss [  1%]
........................ssssssssssssssssssssssssssssssssssssssss........ [  1%]
........................................................................ [  1%]
........................................................................ [  1%]
........................................................................ [  1%]
.....................ssssssssssssssss........................sssssssssss [  1%]
sssssssssssssssssssssssssssss........................................... [  1%]
........................................................................ [  1%]
........................................................................ [  1%]
........................................................................ [  1%]
...........s............................s............................... [  1%]
...................................ssssssssssssssssssssssss............. [  1%]
........................................................................ [  1%]
........................................................................ [  1%]
s...................................ssssssssssssssss.................... [  1%]
....ssssssssssssssssssssssssssssssssssssssss............................ [  1%]
........................................................................ [  2%]
........................................................................ [  2%]
........................................................................ [  2%]
..........................s............................................. [  2%]
..................................................ssssssssssssssssssssss [  2%]
ss...................................................................... [  2%]
........................................................................ [  2%]
...................................................ssssssssssssssss..... [  2%]
...................ssssssssssssssssssssssssssssssssssssssss............. [  2%]
........................................................................ [  2%]
........................................................................ [  2%]
............s.ss.ss..................FF.........s....................... [  2%]
........................................................................ [  2%]
........................................................................ [  2%]
........................................................................ [  2%]
.............................s................s.s....................... [  2%]
........................................................................ [  2%]
........................................................................ [  2%]
........................................................................ [  2%]
........................................................................ [  2%]
........................................................................ [  2%]
........................................................................ [  2%]
...................................s.................................... [  2%]
........................................................................ [  2%]
.........s...............F.............................................. [  2%]
........................................................................ [  2%]
........................................................................ [  2%]
........................................................................ [  3%]
........................................................................ [  3%]
........................................................................ [  3%]
....................s................................................... [  3%]
............................................ssssssssssssssssssssssss.... [  3%]
........................................................................ [  3%]
........................................................................ [  3%]
.............................................ssssssssssssssss........... [  3%]
.............ssssssssssssssssssssssssssssssssssssssss................... [  3%]
.............................................s.......................... [  3%]
..s..................................................................... [  3%]
........................................................................ [  3%]
........................................................................ [  3%]
...........................................................s............ [  3%]
........................................................................ [  3%]
...........ssssssssssssssssssssssss..................................... [  3%]
........................................................................ [  3%]
........................................................................ [  3%]
............ssssssssssssssss........................ssssssssssssssssssss [  3%]
ssssssssssssssssssss.................................................... [  3%]
............ssssssssssssssssssssssss.................................... [  3%]
........................................................................ [  3%]
.................................................s...................... [  3%]
.............ssssssssssssssss........................sssssssssssssssssss [  3%]
sssssssssssssssssssss................................................... [  3%]
.............................ss....................ss...sssss........... [  3%]
......ss................................................................ [  3%]
..............ss..................ss...ss..............sssssssssssssssss [  4%]
sssss.................................................ss................ [  4%]
.........s........s..................................................... [  4%]
..ssssssssssssssssssssssss..........................................ss.. [  4%]
....s..............................................ssssssssssssssssss... [  4%]
........................................................................ [  4%]
.......ss...xxx......................................................... [  4%]
........................................................................ [  4%]
...........s.........s...........s.....s.....s......s.........s......... [  4%]
s............sss........s......s.........s...x..s.........s.........s... [  4%]
......s..........ssssssssssssssssssssssss......s.........s...........s.. [  4%]
.........s...........s.........s.........s.........s.........s.........s [  4%]
...........ss...xxxx.........s...........s...........s.................. [  4%]
................ssssss.................................................. [  4%]
......x......x.........s....x...........x.......................s.s...x. [  4%]
....s.ss....x...xss...............s.....ss...sssss.....sssssssssssssssss [  4%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss..sssss [  4%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssss              [  4%]
test_bundled_inputs.py ............                                      [  4%]
test_comparison_utils.py .....                                           [  4%]
test_complex.py ....                                                     [  4%]
test_cpp_extensions_jit.py F.FFFsFsssF.FFFsssFFFFF                       [  4%]
test_cpp_extensions_open_device_registration.py F                        [  4%]
test_cuda.py FFFFFFsFFFssssFFFFFFFFFFFsFFFsFFFFssssFssFsFssssssFssssFFFs [  4%]
FFsFFFFFFsssFsFFFFFFFssFFsFFsFFFFssFFFFFFFFFFFsFsFssssssssssssFFFF.sFFsF [  4%]
FsssssssssssssssssFsFsFEEEEsFssssssssssFFFFssssFssFFFFFF                 [  4%]
test_cuda_nvml_based_avail.py EEEEEEFFF                                  [  4%]
test_cuda_primary_ctx.py sss                                             [  4%]
test_cuda_sanitizer.py FFFFFFFFFFFFFFFFFFFFFFFFFFFFFF                    [  4%]
test_cuda_trace.py FFFFFFFFFFFF                                          [  4%]
test_dataloader.py ..................................................... [  4%]
......s..........s......s.......Fssss................................... [  4%]
.......s..........s......s........ss...                                  [  4%]
test_datapipe.py ..................sssssss..........................ss.s [  4%]
.sss......sss......................                                      [  4%]
test_decomp.py ......................................................... [  4%]
.............FF......................................................... [  4%]
...FFFF..*** Error in `/gpfs02/astro/workarea/beckermr/miniconda3/envs/pytorch-test/bin/python3.10': free(): invalid next size (fast): 0x000055e1589b6900 ***
======= Backtrace: =========
```

</details>

MKL:

<details/>

```bash
============================= test session starts ==============================
platform linux -- Python 3.11.4, pytest-7.4.0, pluggy-1.2.0
rootdir: /direct/astro+u/beckermr/pytorch-mkl
configfile: pytest.ini
plugins: hypothesis-6.82.0
collected 192953 items / 60 errors / 1 skipped

run_test.py EEEE                                                         [  0%]
test_ao_sparsity.py .[W qlinear_dynamic.cpp:247] Warning: Currently, qnnpack incorrectly ignores reduce_range when it is set to true; this may change in a future release. (function operator())
................................................... [  0%]
.................................                                        [  0%]
test_autocast.py .......s.                                               [  0%]
test_autograd.py .................................s.s......sss.......... [  0%]
........................................................................ [  0%]
............s..............................x............................ [  0%]
....F.F................................................................. [  0%]
........................................................................ [  0%]
......................ss................................................ [  0%]
.....................ss................................................. [  0%]
.s.s....ss.s...............s....sss.s..s........s.......s.....ss...      [  0%]
test_binary_ufuncs.py .................................................. [  0%]
........................................................................ [  0%]
........................................................................ [  0%]
........................................................................ [  0%]
........................................................................ [  0%]
..................s..................................................... [  0%]
........................................................................ [  0%]
........................................................................ [  0%]
.......................s................................................ [  0%]
...............................................ssssssssssssssssssssssss. [  0%]
........................................................................ [  0%]
........................................................................ [  0%]
................................................ssssssssssssssss........ [  0%]
................ssssssssssssssssssssssssssssssssssssssss................ [  0%]
.....................................s.................................. [  0%]
ssss.................................................................... [  0%]
.......................................F................................ [  0%]
........................................................................ [  0%]
........................................................................ [  1%]
.....................................................s.................. [  1%]
........................................................................ [  1%]
.....ssssssssssssssssssssssss........................................... [  1%]
........................................................................ [  1%]
........................................................................ [  1%]
...............................s........................................ [  1%]
.......................................................sssssssssssssssss [  1%]
sssssss................................................................. [  1%]
........................................................................ [  1%]
........................................................ssssssssssssssss [  1%]
........................ssssssssssssssssssssssssssssssssssssssss........ [  1%]
........................................................................ [  1%]
........................................................................ [  1%]
........................................................................ [  1%]
.....................ssssssssssssssss........................sssssssssss [  1%]
sssssssssssssssssssssssssssss........................................... [  1%]
........................................................................ [  1%]
........................................................................ [  1%]
........................................................................ [  1%]
...........s............................s............................... [  1%]
...................................ssssssssssssssssssssssss............. [  1%]
........................................................................ [  1%]
........................................................................ [  1%]
s...................................ssssssssssssssss.................... [  1%]
....ssssssssssssssssssssssssssssssssssssssss............................ [  1%]
........................................................................ [  2%]
........................................................................ [  2%]
........................................................................ [  2%]
..........................s............................................. [  2%]
..................................................ssssssssssssssssssssss [  2%]
ss...................................................................... [  2%]
........................................................................ [  2%]
...................................................ssssssssssssssss..... [  2%]
...................ssssssssssssssssssssssssssssssssssssssss............. [  2%]
........................................................................ [  2%]
........................................................................ [  2%]
............s.ss.ss..................FF.........s....................... [  2%]
........................................................................ [  2%]
........................................................................ [  2%]
........................................................................ [  2%]
.............................s................s.s....................... [  2%]
........................................................................ [  2%]
........................................................................ [  2%]
........................................................................ [  2%]
........................................................................ [  2%]
........................................................................ [  2%]
........................................................................ [  2%]
...................................s.................................... [  2%]
........................................................................ [  2%]
.........s...............F.............................................. [  2%]
........................................................................ [  2%]
........................................................................ [  2%]
........................................................................ [  3%]
........................................................................ [  3%]
........................................................................ [  3%]
....................s................................................... [  3%]
............................................ssssssssssssssssssssssss.... [  3%]
........................................................................ [  3%]
........................................................................ [  3%]
.............................................ssssssssssssssss........... [  3%]
.............ssssssssssssssssssssssssssssssssssssssss................... [  3%]
.............................................s.......................... [  3%]
..s..................................................................... [  3%]
........................................................................ [  3%]
........................................................................ [  3%]
...........................................................s............ [  3%]
........................................................................ [  3%]
...........ssssssssssssssssssssssss..................................... [  3%]
........................................................................ [  3%]
........................................................................ [  3%]
............ssssssssssssssss........................ssssssssssssssssssss [  3%]
ssssssssssssssssssss.................................................... [  3%]
............ssssssssssssssssssssssss.................................... [  3%]
........................................................................ [  3%]
.................................................s...................... [  3%]
.............ssssssssssssssss........................sssssssssssssssssss [  3%]
sssssssssssssssssssss................................................... [  3%]
.............................ss....................ss...sssss........... [  3%]
......ss................................................................ [  3%]
..............ss..................ss...ss..............sssssssssssssssss [  4%]
sssss.................................................ss................ [  4%]
.........s........s..................................................... [  4%]
..ssssssssssssssssssssssss..........................................ss.. [  4%]
....s..............................................ssssssssssssssssss... [  4%]
........................................................................ [  4%]
.......ss...xxx......................................................... [  4%]
........................................................................ [  4%]
...........s.........s...........s.....s.....s......s.........s......... [  4%]
s............sss........s......s.........s...x..s.........s.........s... [  4%]
......s..........ssssssssssssssssssssssss......s.........s...........s.. [  4%]
.........s...........s.........s.........s.........s.........s.........s [  4%]
...........ss...xxxx.........s...........s...........s.................. [  4%]
................ssssss.................................................. [  4%]
......x......x.........s....x...........x.......................s.s...x. [  4%]
....s.ss....x...xss...............s.....ss...sssss.....sssssssssssssssss [  4%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss..sssss [  4%]
sssssssssssssssssssssssssssssssssssssssssssssssssssssssssss              [  4%]
test_bundled_inputs.py ............                                      [  4%]
test_comparison_utils.py .....                                           [  4%]
test_complex.py ....                                                     [  4%]
test_cpp_extensions_jit.py F.FFFsFsssF.FFFsssFFFFF                       [  4%]
test_cpp_extensions_open_device_registration.py F                        [  4%]
test_cuda.py FFFFFFsFFFssssFFFFFFFFFFFsFFFsFFFFssssFssFsFssssssFssssFFFs [  4%]
FFsFFFFFFsssFsFFFFFFFssFFsFFsFFFFssFFFFFFFFFFFsFsFssssssssssssFFFF.sFFsF [  4%]
FsssssssssssssssssFsFsFEEEEsFssssssssssFFFFssssFssFFFFFF                 [  4%]
test_cuda_nvml_based_avail.py EEEEEEFFF                                  [  4%]
test_cuda_primary_ctx.py sss                                             [  4%]
test_cuda_sanitizer.py FFFFFFFFFFFFFFFFFFFFFFFFFFFFFF                    [  4%]
test_cuda_trace.py FFFFFFFFFFFF                                          [  4%]
test_dataloader.py ..................................................... [  4%]
......s..........s......s.......Fssss....F.............................. [  4%]
.......s..........s......s........ss...                                  [  4%]
test_datapipe.py ..................sssssss..........................ss.s [  4%]
.sss......sss......................                                      [  4%]
test_decomp.py ......................................................... [  4%]
........................................................................ [  4%]
...FFFF..*** Error in `/gpfs02/astro/workarea/beckermr/miniconda3/envs/pytorch-test-mkl/bin/python3.11': free(): invalid next size (fast): 0x000056228fbe3500 ***
======= Backtrace: =========

```

</details>

**Comment by beckermr** [comment]
Overall, I'd say it's fine to merge this PR. The openblas builds aren't any worse than the MKL ones in any significant way AFAICT.

**Comment by jakirkham** [comment]
Thanks all! 🙏

Nice to see we can choose the BLAS used with PyTorch 😀

Am curious, what is still needed in this PR (if anything)?

**Comment by beckermr** [comment]
Nothing afaik. Waiting for @rgommers and/or @isuruf to comment before merging.

**Comment by beckermr** [comment]
The segfaults are in the same spot for the current builds on anaconda.org

**Comment by beckermr** [comment]
The cuda builds have not changed so need to remake them. We don't ship cuda + openblas.

**Comment by beckermr** [comment]
I'm happy to hold off. AFAICT most/all of the PyTorch 2.0 builds are broken for various reasons.

**Comment by hmaarrfk** [comment]
I'm ok merging. This doesn't increase my personal maintenance burden because the builds seem to complete on the CIs

**Comment by rgommers** [comment]
> Waiting for @rgommers and/or @isuruf to comment before merging.

Sorry, didn't mean to hold this up - had posted a 👍🏼 already. Thanks for doing the due diligence. I'll see if I can run the test suite locally too and get a bit further than 4% on either of the builds.

**Comment by beckermr** [comment]
Ahh I missed the thumbs up! That's my fault. Let's wait for @ngam until the end of the day or so and then merge.

**Comment by jakirkham** [comment]
Thanks all! 🙏

**Comment by jakirkham** [comment]
> We don't ship cuda + openblas

Raised issue ( https://github.com/conda-forge/pytorch-cpu-feedstock/issues/177 ) to discuss how we might enable CUDA with different BLAS implementations as well

**Pull request merged**